fix(plugin test): close connection in plugin test environment

zryanl opened the pull request:
- Addresses a connection leak in the internal plugin test environment's network connection instantiation
- Implements some extra error handling and checks in the plugin test package

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Kong/go-pdk?pullRequest=211) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Kong/go-pdk?pullRequest=211) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Kong/go-pdk?pullRequest=211) it.</sub>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Kong/go-pdk?pullRequest=211) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Kong/go-pdk?pullRequest=211) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Kong/go-pdk?pullRequest=211) it.</sub>

gszr commented on bridge/bridgetest/bridgetest.go:
```diff
@@ -20,7 +20,7 @@ func readPbFrame(conn net.Conn) (data []byte, err error) {
 	err = binary.Read(conn, binary.LittleEndian, &len)
 	if err != nil {
-		return
+		return nil, err
```
Note that these and others below make use of [named return values](https://go.dev/tour/basics/7), so these changes have no effect.

gszr commented on test/test.go:
```diff
@@ -122,19 +122,23 @@ func (req *Request) Validate() error {
 		}
 		return nil
 	}
-	return fmt.Errorf("Unsupported method \"%v\"", req.Method)
```
Let's not change this error message.

gszr commented on test/test.go:
```diff
@@ -122,19 +122,23 @@ func (req *Request) Validate() error {
 	}
-	portnum, _ := strconv.Atoi(p)
-	return int32(portnum)
+	portnum, err := strconv.ParseInt(p, 10, 32)
```
The port is already validated in the `URL` object creation, so this should be unreachable.

gszr commented on test/test.go:
```diff
@@ -196,14 +200,14 @@ type TestEnv struct {
 	err = req.Validate()
 	if err != nil {
-		return
+		return nil, err
```
Named return value. Let's not change this and others.

gszr reviewed (changes requested):
@zryanl Thanks for your contribution; please restrict this PR to only the close connection fix and I will be happy to re-review.

zryanl committed:
fix(plugin test): close connection in plugin test environment

- Addresses a connection leak in the internal plugin test environment's network connection instantiation

zryanl committed:
tests(util): add helper for testing multiple plugins

zryanl commented:
Hi @gszr I believe I've addressed your comment, are there any other outstanding with the PR?
